scripts/install_python_dev_dependencies.py

```diff
@@ -0,0 +1,137 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Install Python development dependencies."""
+
+from __future__ import annotations
+
+import argparse
+import os
+import subprocess
+import sys
+
+from typing import List, Optional
+
+
+INSTALLATION_TOOL_VERSIONS = {
+    'pip': '22.1.1',
+    'pip-tools': '6.6.2',
+    'setuptools': '58.5.3',
+}
+REQUIREMENTS_DEV_FILE_PATH = 'requirements_dev.in'
+COMPILED_REQUIREMENTS_DEV_FILE_PATH = 'requirements_dev.txt'
+
+_PARSER = argparse.ArgumentParser(
+    'Install Python development dependencies')
+_PARSER.add_argument(
+    '--assert_compiled', action='store_true',
+    help='Assert that the dev requirements file is already compiled.')
+
+
+def assert_in_venv() -> None:
+    """Raise an error if we are not in a virtual environment.
+
+    No error is raised if we are running on GitHub Actions because a
```

Comment:
Given the name of this function, shouldn't the "if GITHUB_ACTION" exception be in the caller of this function instead? (Basically -- if we're not in an Actions environment, don't assert.)

Reply:
done (renamed function)